compiler/generator.ml

```diff
@@ -0,0 +1,57 @@
+(*
+ * COMS4115: Odds parser
+ *
+ * Authors:
+ *  - Alex Kalicki
+ *  - Alexandra Medway
+ *  - Daniel Echikson
+ *  - Lilly Wang
+ *)
+
+open Ast
+open Printf
+
+let rec txt_of_expr expr = match expr with
+  | Int_lit(i) -> string_of_int(i)
+  | Float_lit(f) -> string_of_float(f)
+  | String_lit(s) -> sprintf "\"%s\"" s
+  | Id(e) -> e
+  | Binop(e1, op, e2) ->
+      sprintf "%s %s %s" (txt_of_expr e1) (txt_of_op op) (txt_of_expr e2)
+  | Unop(op, e) -> sprintf "%s %s" (txt_of_op op) (txt_of_expr e)
```

Comment:
we shouldn't have a space between the two strings here right? e.g. `Unop(Minus, 5)` should evaluate to `-5` not `- 5`